src/lints/function_export_name_changed.ron

```diff
@@ -0,0 +1,47 @@
+SemverQuery(
+    id: "function_export_name_changed",
+    human_readable_name: "function's export name has changed or been removed",
+    description: "A function's ABI name with #[no_mangle] or #[export_name = \"name\"] has changed or been removed",
+    required_update: Major,  // Not specified in rust semver doc? most similar is for #[repr] changes: https://doc.rust-lang.org/cargo/reference/semver.html#type-layout
+    reference_link: Some("https://doc.rust-lang.org/reference/abi.html#the-no_mangle-attribute"),  // TODO
+    query: r#"
+    {
+        CrateDiff {
+            baseline {
+                item {
+                    ... on Function {
+                        importable_path {
+                            path @output @tag
+                        }
+
+                        export_name @output @tag @filter(op: "is_not_null")
+                    }
+                }
+            }
+            
+            current {
+                item {
+                    ... on Function {
+                        name @output
+
+                        export_name @filter(op: "!=", value: ["%export_name"])
```

Comment:
For the sake of easier debugging, let's capture for output the new export name here. That way the new name will get recorded in the snapshot files and will be available in `--verbose` runs, which simplifies debugging if users report issues with the lint:
```suggestion
                        new_export_name: export_name @filter(op: "!=", value: ["%export_name"]) @output
```

You will probably need to recreate the snapshot test data, or just add the extra column manually.